tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -357,25 +390,64 @@ private void AsmDataProductConfigChanged(object sender, ProductConfigChangedEven
             }
         }
 
-        private void AsmProductConfigChanged(object sender, ProductConfigChangedEventArgs e)
+        private void AsmProductConfigRemoved(object sender, ProductConfigChangedEventArgs e)
         {
             if (!_enabled) { return; }
 
             var asmConfigs = e.GetDeserializedConfigurations<RcmModels.Asm.Payload>();
 
-            _remoteConfigurationStatus.RulesOverrides.Clear();
-            _remoteConfigurationStatus.Exclusions.Clear();
+            foreach (var asmConfig in asmConfigs)
+            {
+                if (_remoteConfigurationStatus.RulesOverridesByFile.ContainsKey(asmConfig.Name))
+                {
+                    _remoteConfigurationStatus.RulesOverridesByFile.Remove(asmConfig.Name);
+                }
+
+                if (_remoteConfigurationStatus.ExclusionsByFile.ContainsKey(asmConfig.Name))
+                {
+                    _remoteConfigurationStatus.ExclusionsByFile.Remove(asmConfig.Name);
+                }
+            }
```

Comment:
also in this block should we handle `actions` the same way? seems weird to have several config files for it... but theoretically could happen that they're reset by removing the file rather than sending an empty list🤔